src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -208,23 +208,34 @@ public void SendFunctionLoadRequests()
 
         public void SendFunctionEnvironmentReloadRequest()
         {
+            _workerChannelLogger.LogDebug("Sending SendFunctionEnvironmentReloadRequest");
             _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionEnvironmentReloadResponse)
       .Subscribe((msg) => PublishWorkerProcessReadyEvent(msg.Message.FunctionEnvironmentReloadResponse)));
 
             IDictionary processEnv = Environment.GetEnvironmentVariables();
 
-            FunctionEnvironmentReloadRequest request = new FunctionEnvironmentReloadRequest();
-            foreach (DictionaryEntry entry in processEnv)
-            {
-                request.EnvironmentVariables.Add(entry.Key.ToString(), entry.Value.ToString());
-            }
+            FunctionEnvironmentReloadRequest request = GetFunctionEnvironmentReloadRequest(processEnv);
 
             SendStreamingMessage(new StreamingMessage
             {
                 FunctionEnvironmentReloadRequest = request
             });
         }
 
+        internal FunctionEnvironmentReloadRequest GetFunctionEnvironmentReloadRequest(IDictionary processEnv)
+        {
+            FunctionEnvironmentReloadRequest request = new FunctionEnvironmentReloadRequest();
+            foreach (DictionaryEntry entry in processEnv)
+            {
+                if (entry.Value != null && !string.IsNullOrEmpty(entry.Value.ToString()))
+                {
+                    request.EnvironmentVariables.Add(entry.Key.ToString(), entry.Value.ToString());
```

Comment:
Should we special case this for node? This might result in breaking behavior for current Java users since technically an environment variable value can be empty, it's just a node grpc quirk that the "map<string, string>" type we chose requires non-empty values

Reply:
you are right. Will update to sanitize only if the worker has requests it in the capabilities

Reply:
Verfied that [Environment.GetEnvironmentVariables](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariables?view=netcore-2.2) does not include keys with empty or null values. No action needed. Updated the PR to add a test,